File: src/main/java/com/baidu/algorithm/binarysearch/_540_Single_Element_in_a_Sorted_Array.py
class Solution(object):
    def singleNonDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        left, right = 0, n / 2
        while left < right:
            m = (left + right) >> 1
            if nums[2 * m] == nums[2 * m + 1]:
                left = m + 1
            else:
                right = m

        return nums[2 * left]

File: src/main/java/com/baidu/algorithm/binarysearch/_540_Single_Element_in_a_Sorted_Array.java
/*
 * Copyright (C) 2018 Baidu, Inc. All Rights Reserved.
 */
package com.baidu.algorithm.binarysearch;

/**
 * _540_Single_Element_in_a_Sorted_Array
 *
 * @author xuhaoran01
 */
public class _540_Single_Element_in_a_Sorted_Array {
    public int singleNonDuplicate(int[] nums) {
        if (nums == null || nums.length == 0) {
            return -1;
        }

        int left = 0, right = nums.length - 1;
        while (left < right) {
            int mid = (left + right) / 2;
            if (mid % 2 == 1) {
                mid--;
            }

            if (nums[mid] == nums[mid + 1]) {
                left = mid + 2;
            } else {
                right = mid;
            }
        }

        return nums[left];
    }
}
